x-pack/plugins/security_solution/server/lib/detection_engine/notifications/legacy_saved_object_references/legacy_extract_references.ts

```diff
@@ -0,0 +1,62 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from 'src/core/server';
+import { RuleParamsAndRefs } from '../../../../../../alerting/server';
+// eslint-disable-next-line no-restricted-imports
+import { LegacyRulesNotificationParams } from '../legacy_types';
+// eslint-disable-next-line no-restricted-imports
+import { legacyExtractRuleId } from './legacy_extract_rule_id';
+
+/**
+ * Extracts references and returns the saved object references.
+ * NOTE: You should not have to add any new ones here at all, but this keeps consistency with the other
+ * version(s) used for security_solution rules.
+ *
+ * How to add a new extracted references here (This should be rare or non-existent):
+ * ---
+ * Add a new file for extraction named: extract_<paramName>.ts, example: extract_foo.ts
+ * Add a function into that file named: extract<ParamName>, example: extractFoo(logger, params.foo)
+ * Add a new line below and concat together the new extract with existing ones like so:
+ *
+ * const legacyRuleIdReferences = legacyExtractRuleId(logger, params.ruleAlertId);
+ * const fooReferences = extractFoo(logger, params.foo);
+ * const returnReferences = [...legacyRuleIdReferences, ...fooReferences];
+ *
+ * Optionally you can remove any parameters you do not want to store within the Saved Object here:
+ * const paramsWithoutSavedObjectReferences = { removeParam, ...otherParams };
+ *
+ * If you do remove params, then update the types in: security_solution/server/lib/detection_engine/notifications/legacy_rules_notification_alert_type.ts
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ * @param logger Kibana injected logger
+ * @param params The params of the base rule(s).
+ * @returns The rule parameters and the saved object references to store.
+ */
+export const legacyExtractReferences = ({
+  logger,
+  params,
+}: {
+  logger: Logger;
+  params: LegacyRulesNotificationParams;
+}): RuleParamsAndRefs<LegacyRulesNotificationParams> => {
+  const legacyRuleIdReferences = legacyExtractRuleId({
+    logger,
+    ruleAlertId: params.ruleAlertId,
+  });
+  const returnReferences = [...legacyRuleIdReferences];
+
+  // Modify params if you want to remove any elements separately here. For legacy ruleAlertId, we do not remove the id and instead
+  // keep it to both fail safe guard against manually removed saved object references or if there are migration issues and the saved object
+  // references are removed. Also keeping it we can detect and log out a warning if the reference between it and the saved_object reference
+  // array have changed between each other indicating the saved_object array is being mutated outside of this functionality
+  const paramsWithoutSavedObjectReferences = { ...params };
```

Comment:
same comment as above...do you need this destructuring?

Reply:
Not technically, but it does give us a shallow copy of the params. We might end up removing the extra parameters as indicated in the docs before shipping and changing this slightly. This was mostly illustrative and inline with our other one.

We might change this slightly after a team discussion about how much "safety" we want where we aren't removing the parameters that we get from saved object references just yet, but we probably should just go ahead and remove the parameters we aren't using anymore to avoid programmers using them instead of what's in the saved object references. If we do remove them that's where we would remove them with the destructoring. 

I will talk to @dhurley14 and @yctercero about if we want to remove the unused parameters now that we are far enough along and feel comfortable enough.

Reply:
But then again, ew aren't planning on new ones being created since it's deprecated. Really the only time this should be executed or run is only whenever we are doing testing locally as we took out the functionality outside of testing for users to add new legacy notifications.

Reply:
I think it's fine to leave for now. If anything after testing all these PRs together, we can add this as followup/cleanup.